include/p4mlir/Dialect/P4HIR/P4HIR_Attrs.td

```diff
@@ -27,6 +27,9 @@ def P4HIR_BoolAttr : P4HIR_Attr<"Bool", "bool", [TypedAttrInterface]> {
     AttrBuilder<(ins "bool":$value), [{
       return $_get($_ctxt, BoolType::get($_ctxt), value);
     }]>,
+    AttrBuilderWithInferredContext<(ins "mlir::Type":$type, "bool":$value), [{
```

Comment:
This is nonsense. `BoolAttr` could only be of bool type.

Reply:
context is the object that owns types (and attributes). Sometimes we can deduce it from somewhere else (e.g. from compound types we certainly need to use the same context as the inner type). So, the builder w/o context is just a sugar, in general we should always have a build with context available.